pkg/types/validation/installconfig.go

```diff
@@ -329,6 +329,10 @@ func validateNetworking(n *types.Networking, singleNodeOpenShift bool, fldPath *
 		allErrs = append(allErrs, field.Required(fldPath.Child("networkType"), "network provider type required"))
 	}
 
+	if n.NetworkType == string(operv1.NetworkTypeKuryr) {
```

Comment:
Shouldn't we have a list of valid network types coming from the API, and error when the user provides an unknown network type? If so, we should start by removing the Kuryr network type from the API, then simply bump the installer to the new API.

Reply:
There is no such validation in the installer. I've just started installation with `networking.type="Foobar" it will only fail on CNO level. With this patch I want to clean up Kuryr bits, I don't want to introduce new validations that are not related to Kuryr.

Reply:
Do we really need this validation then, if the installer happily accepts any network type?
We don't want to depend on `NetworkTypeKuryr` as we're removing it from the API.

Reply:
My point with this is to give users feedback on the installer level that Kuryr is not supported anymore. Otherwise installer will happily accept `networkType=Kuryr` and user will only know that this doesn't work once network-operator gets up and reports (or not) an error. How about we change that to a hardcoded string to anticipate the API removal?

Reply:
That works for me. At some point, it would also be good to add a more generic validation to the installer, so that users with fat fingers don't spend hours debugging their failed deployments.

Reply:
Done, thanks!